examples/content_types/src/tests.rs

```diff
@@ -0,0 +1,52 @@
+use super::rocket;
+use super::serde_json;
+use super::Person;
+use rocket::http::{ContentType, Method, Status};
+use rocket::testing::MockRequest;
+
+fn test(uri: &str,
+        content_type: ContentType,
+        expected_status: Status,
+        expected_body: String) {
+    let rocket = rocket::ignite()
+        .mount("/hello", routes![super::hello])
+        .catch(errors![super::not_found]);
+    let mut request = MockRequest::new(Method::Get, uri).header(content_type);
+    let mut response = request.dispatch_with(&rocket);
+
+    assert_eq!(response.status(), expected_status);
+    assert_eq!(response.body().and_then(|b| b.into_string()), Some(expected_body));
+}
+
+#[test]
+fn test_hello() {
+    let person = Person {
+        name: "Michael".to_string(),
+        age: 80,
+    };
+
+    test("/hello/Michael/80",
+         ContentType::JSON,
+         Status::Ok,
+         serde_json::to_string(&person).unwrap());
+}
+
+#[test]
+fn test_hello_invalid_content_type() {
+    test("/hello/Michael/80",
+         ContentType::HTML,
+         Status::NotFound,
+         format!("<p>This server only supports JSON requests, not '{}'.</p>",
+                 ContentType::HTML));
+}
+
+#[test]
+fn test_404() {
+    test("/unknown",
+         ContentType::JSON,
+         Status::NotFound,
+         "<p>Sorry, '/unknown' is not a valid path!</p>
```

Comment:
This is pretty mysterious spacing, especially with the rogue '/' in there. Consider changing the actual route somehow to make this clearer.